ui/components/MesheryPerformance/Dashboard.js

```diff
@@ -43,18 +44,28 @@ function Dashboard({
   function fetchTestProfiles() {
     updateProgress({ showProgress : true });
 
-    dataFetch(
-      `${MESHERY_PERFORMANCE_URL}`,
-      { credentials : "include", },
-      (result) => {
-        updateProgress({ showProgress : false });
-        if (result) {
-          setProfiles({ count : result.total_count || 0,
-            profiles : result.profiles || [], });
+    fetchPerformanceProfiles({
+      selector : {
+        // default
+        pageSize : `10`,
+        page : `0`,
+        search : ``,
+        order : ``,
+      },
+    }).subscribe({
+      next : (res) => {
+        // @ts-ignore
+        let result = res?.getPerformanceProfiles;
+        if (typeof result !== "undefined") {
+          updateProgress({ showProgress : false });

```

Comment:
This can be moved outside of the if block ig, because, if the `result` is undefined, then the progress bar will not stop, which can be frustrating at times 😅